src/listeners/cors_listener.cr

```diff
@@ -10,6 +10,68 @@ struct Athena::Routing::Listeners::CORS
   # :nodoc:
   ALLOW_SET_ORIGIN = "athena.routing.cors.allow_set_origin"
   private WILDCARD         = "*"
+
+  # Encapsulates logic to set CORS response headers
+  struct ResponseHeaders
+    def initialize(@headers : HTTP::Headers)
+    end
+
+    {% for header in %w[allow-origin allow-methods allow-headers allow-credentials expose-headers] %}
+      {% method_name = header.tr("-", "_").id %}
+      def {{method_name}}=(value : String)
```

Comment:
```suggestion
      def {{method_name}}=(value : String) : Nil
```

Minor, but I like to declare argument/return types where possible.  Just makes the intended behavior a bit more clear for future readers.

Same thing for the other methods below.